pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -477,6 +477,408 @@ def default_controlled_resource_decomp(
         raise re.ResourcesNotDefined
 
 
+class ResourceControlledSequence(ResourceOperator):
+    r"""Resource class for the ControlledSequence gate.
+
+    This operator represents a sequence of controlled gates, one for each control wire, with the
+    base operator (:code:`base_op`) raised to decreasing powers of 2.
+
+    Args:
+        base_op (~.pennylane.labs.resource_estimation.ResourceOperator): The operator that we
+            will be applying controlled powers of.
+        num_control_wires (int): the number of controlled wires to run the sequence over
+        wires (Sequence[int], optional): the wires the operation acts on
```

Comment:
Will be great to be consistent with qml.ControlledSequence in terms of naming, description etc. Same for other operators.